tests/unit/workflows/nodejs_npm/test_actions.py

```diff
@@ -78,3 +79,63 @@ def test_raises_action_failed_when_npm_fails(self, SubprocessNpmMock):
             action.execute()
 
         self.assertEqual(raised.exception.args[0], "NPM Failed: boom!")
+
+
+class TestNodejsNpmrcCopyAction(TestCase):
+
+    @patch("aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils")
+    def test_copies_npmrc_into_a_project(self, OSUtilMock):
+        osutils = OSUtilMock.return_value
+        osutils.joinpath.side_effect = lambda a, b: "{}/{}".format(a, b)
+
+        action = NodejsNpmrcCopyAction("artifacts",
+                                       "source",
+                                       osutils=osutils)
+        osutils.file_exists.side_effect = [True]
+        action.execute()
+
+        osutils.file_exists.assert_called_with("source/.npmrc")
+        osutils.copy_file.assert_called_with("source/.npmrc", "artifacts")
+
+    @patch("aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils")
+    def test_skips_copying_npmrc_into_a_project_if_npmrc_doesnt_exist(self, OSUtilMock):
+        osutils = OSUtilMock.return_value
+        osutils.joinpath.side_effect = lambda a, b: "{}/{}".format(a, b)
+
+        action = NodejsNpmrcCopyAction("artifacts",
+                                       "source",
+                                       osutils=osutils)
+        osutils.file_exists.side_effect = [False]
+        action.execute()
+
+        osutils.file_exists.assert_called_with("source/.npmrc")
+        osutils.copy_file.assert_not_called()
+
+    @patch("aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils")
+    def test_raises_action_failed_when_copying_fails(self, OSUtilMock):
+        osutils = OSUtilMock.return_value
+        osutils.joinpath.side_effect = lambda a, b: "{}/{}".format(a, b)
+
+        osutils.copy_file.side_effect = OSError()
+
+        action = NodejsNpmrcCopyAction("artifacts",
+                                       "source",
+                                       osutils=osutils)
+
+        with self.assertRaises(ActionFailedError):
+            action.execute()
+
+
+class TestNodejsCleanUpAction(TestCase):
+
+    @patch("aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils")
+    def test_removes_npmrc_if_npmrc_exists(self, OSUtilMock):
```

Comment:
Can you also add a tests for when npmrc does not exist, to make sure that branch is also covered?

Reply:
sure, I'll take a look at it

Reply:
done.